src/agent/onefuzz-task/src/tasks/config.rs

```diff
@@ -364,3 +384,79 @@ impl Config {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use onefuzz::expand::{GetExpand, PlaceHolder};
+    use proptest::prelude::*;
+
+    use crate::config_test_utils::GetExpandFields;
+
+    use super::CommonConfig;
+
+    impl GetExpandFields for CommonConfig {
+        fn get_expand_fields(&self) -> Vec<(PlaceHolder, String)> {
+            let mut params = vec![
+                (
+                    PlaceHolder::MachineId,
+                    self.machine_identity.machine_id.to_string(),
+                ),
+                (PlaceHolder::JobId, self.job_id.to_string()),
+                (PlaceHolder::TaskId, self.task_id.to_string()),
+                (
+                    PlaceHolder::SetupDir,
+                    dunce::canonicalize(&self.setup_dir)
+                        .unwrap()
+                        .to_string_lossy()
+                        .to_string(),
+                ),
+            ];
+            if let Some(key) = &self.instance_telemetry_key {
+                params.push((PlaceHolder::InstanceTelemetryKey, key.to_string()));
+            }
+            if let Some(key) = &self.microsoft_telemetry_key {
+                params.push((PlaceHolder::MicrosoftTelemetryKey, key.clone().to_string()));
+            }
+            if let Some(dir) = &self.extra_setup_dir {
+                params.push((
+                    PlaceHolder::ExtraSetupDir,
+                    dunce::canonicalize(dir)
+                        .unwrap()
+                        .to_string_lossy()
+                        .to_string(),
+                ));
+            }
+            if let Some(dir) = &self.extra_output {
+                params.push((
+                    PlaceHolder::ExtraOutputDir,
+                    dunce::canonicalize(&dir.local_path)
+                        .unwrap()
+                        .to_string_lossy()
+                        .to_string(),
+                ));
+            }
+
+            params
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn test_get_expand_values_match_config(
+            config in any::<CommonConfig>(),
+        ) {
+            // This function implementation is repeated across all config tests
+            // There might be a way to share it by taking advantage of the `GetExpandFields` trait, but I'm not sure how
```

Comment:
You could use a macro that takes the config type as an input (in this case: `CommonConfig`) and spits out this `proptest!` macro as an output.